tests/virtual_machines_test.go

```diff
@@ -30,9 +32,34 @@ var _ = Describe("Virtual Machines", func() {
 			Body([]byte(fmt.Sprintf(linuxBridgeConfCRD, "linux-bridge", TestNamespace))).
 			Do()
 		Expect(result.Error()).NotTo(HaveOccurred())
+
+		By("Verify that there are no VMs in the cluster before starting test suite")
+		beforeAllVMList := &kubevirtv1.VirtualMachineList{}
+		err := testClient.VirtClient.List(context.TODO(), beforeAllVMList, &client.ListOptions{})
+		Expect(err).ToNot(HaveOccurred(), "Should successfully list add VMs")
+		Expect(len(beforeAllVMList.Items)).To(BeZero(), "There should be no VM's in the cluster before running test suite")
 	})
 
 	Context("Check the client", func() {
+		BeforeEach(func() {
+			By("Verify that there are no VMs left from previous tests")
+			currentVMList := &kubevirtv1.VirtualMachineList{}
+			err := testClient.VirtClient.List(context.TODO(), currentVMList, &client.ListOptions{})
+			Expect(err).ToNot(HaveOccurred(), "Should successfully list add VMs")
+			Expect(len(currentVMList.Items)).To(BeZero(), "There should be no VM's in the cluster before a test")
+
+			By("Clear vmWaitConfigMap configMap data before each test")
+			vmWaitConfigMap, err := testClient.KubeClient.CoreV1().ConfigMaps(names.MANAGER_NAMESPACE).Get(names.WAITING_VMS_CONFIGMAP, meta_v1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Should successfully get %s ConfigMap", names.WAITING_VMS_CONFIGMAP))
+
+			By("Clearing vmWaitConfigMap configMap data")
+			//  Clearing the map in-place instead of waiting to garbage-collector.
```

Comment:
Redundant wuth `By`. If you want to keep the detail about garbage collector, you can add it to By